src/main/java/com/liveramp/cascading_ext/flow/JobRecordListener.java

```diff
@@ -20,11 +25,13 @@
 
   private final boolean failOnCounterFetch;
   private final JobPersister persister;
+  private final List<TaskSummary> taskSummaries;
```

Comment:
I don't say this often, but let's actually start with this null instead of an empty list, so it's clear whether you are trying to fetch this before it exists vs there actually being no summaries.